recipe/meta.yaml

```
@@ -9,19 +9,22 @@ source:
   sha256: 7f6b2733bde06ea2575b79a6450d9bd5c38918ff4cb2a355bf22bbe8c86c6bcf
 
 build:
+  # skip py<39 required from upstream
+  skip: True  # [py<39]
   number: 0
   noarch: python
-  script: {{ PYTHON }} -m pip install . --no-deps -vv
+  script: {{ PYTHON }} -m pip install . --no-deps -vv --no-deps --no-build-isolation
 
 requirements:
   host:
-    - python >=3.9
-    - setuptools >=42
-    - setuptools-scm >=7.0
+    - python
+    - setuptools 67.8.0
+    - setuptools-scm 7.1.0
```

```suggestion
    - setuptools
    - setuptools-scm
```
For python packages we can drop exact pinning in host

---

And, please, add `toml`  https://github.com/h5netcdf/h5netcdf/blob/2951cf8ea3174dde5663451484bde0cebaa535d0/pyproject.toml#L5C21-L5C25

---

@skupr-anaconda  is that the latest sutuptools from defaults would be used which we knew it is newer than what upstream says ?

---

Yeah, it will be the latest available from defaults

---

would work though if we are supposed to deliver recipes which could be buildable on 3rd party systems ( that is the very reason we skipping python -  based on upstream only - not our own build capabilities on prefect ) it could fail miserably, so overall I can see some inconsequence's here, correct me if I am wrong please @skupr-anaconda

---

Interesting question @M-Waszkiewicz-Anaconda

---

> And, please, add `toml` https://github.com/h5netcdf/h5netcdf/blob/2951cf8ea3174dde5663451484bde0cebaa535d0/pyproject.toml#L5C21-L5C25

What does that notation in the square brackets mean? Should we in this case always add `toml` as a dependency?

---

It's a notation for an extra dependency added like `setuptools-scm + toml`

---

By the way, I really don't get this dependency on `toml`. If it is indeed necessary as a host dependency, shouldn't the build break if it's not there?